plugins/pv_lxc.c

```diff
@@ -468,99 +442,14 @@ static void pv_setup_default_log(struct pv_platform *p, struct lxc_container *c,
 	}
 }
 
-/*
- * Use only after loading config.
- * Use threshold as 0 to always truncate.
- */
-static void pv_truncate_lxc_log(struct lxc_container *c,
-				const char *platform_name, off_t threshold,
-				const char *key)
-{
-	char logfile_name[PATH_MAX];
-
-	c->get_config_item(c, key, logfile_name, PATH_MAX);
-	if (!strlen(logfile_name)) {
-		/*
-		 * /storage/logs is hardcoded in pv_lxc_log->lxcpath
-		 * and used when lxc.log.file isn't set in lxc.conf of
-		 * the container.
-		 */
-		__pv_paths_pv_log_file(logfile_name, PATH_MAX,
-				       __pv_get_instance()->state->rev,
-				       platform_name, LXC_LOG_FNAME);
-	}
-	if (!threshold)
-		truncate(logfile_name, 0);
-	else {
-		struct stat st;
-		if (!stat(logfile_name, &st)) {
-			if (st.st_size >= threshold)
-				truncate(logfile_name, 0);
-		}
-	}
-}
-
-static struct pv_log_info *
-pv_create_lxc_log(struct pv_platform *p, struct lxc_container *c,
-		  struct pv_logger_config *item_config)
-{
-	struct pv_log_info *pv_log_i = NULL;
-	const char *log_key = NULL;
-	char logger_name[128] = { 0 };
-	const char *lxc_log_key[][2] = { { "lxc", "lxc.log.file" },
-					 { "console", "lxc.console.logfile" },
-					 { NULL, NULL } };
-	int i = 0, j = 0;
-	while (lxc_log_key[j][0]) {
-		bool found = false;
-		const char *key = lxc_log_key[j][0];
-		i = 0;
-		while (item_config->pair[i][0]) {
-			if (!strncmp(item_config->pair[i][0], key,
-				     strlen(key))) {
-				log_key = lxc_log_key[j][1];
-				found = true;
-				break;
-			}
-			i++;
-		}
-		if (found)
-			break;
-		j++;
-	}
-
-	if (!log_key) {
-		printf("Configuration not for lxc/console log"
-		       " Skipping for %s.\n",
-		       __func__);
-		goto out;
-	}
-	snprintf(logger_name, sizeof(logger_name), "%s-%s", p->name,
-		 (strstr(log_key, "lxc.console") ? "console" : "lxc"));
-
-	pv_log_i = __pv_new_log(true, item_config, logger_name);
-
-	if (pv_log_i) {
-		int ret = pv_setup_lxc_log(pv_log_i, p->name, c, log_key);
-		if (ret) {
-			pv_free_lxc_log(pv_log_i);
-			free(pv_log_i);
-			pv_log_i = NULL;
-		}
-	}
-out:
-	return pv_log_i;
-}
-
 void *pv_start_container(struct pv_platform *p, const char *rev,
-			 char *conf_file, void *data)
+			 char *conf_file, int logpipe, void *data)
```

Comment:
isn't this logfd? if i read pipe i would think of tuple.

Reply:
Ok, I think I should unify the name, logfd will be the name